python/cudf_kafka/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+# =============================================================================
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
+
+set(cudf_kafka_version 23.12.00)
+
+include(../../fetch_rapids.cmake)
+include(rapids-cuda)
+rapids_cuda_init_architectures(cudf-kafka-python)
+
+project(
+  cudf-kafka-python
+  VERSION ${cudf_kafka_version}
+  LANGUAGES # TODO: Building Python extension modules via the python_extension_module requires the C
+            # language to be enabled here. The test project that is built in scikit-build to verify
+            # various linking options for the python library is hardcoded to build with C, so until
+            # that is fixed we need to keep C.
+            C CXX CUDA
+)
+
+option(FIND_CUDF_KAFKA_CPP
+       "Search for existing cudf_kafka C++ installations before defaulting to local files" OFF
+)
+
+option(CUDF_KAFKA_BUILD_WHEELS "Whether this build is generating a Python wheel." OFF)
+
+# If the user requested it we attempt to find cudf_kafka.
+if(FIND_CUDF_KAFKA_CPP)
+  find_package(cudf_kafka ${cudf_kafka_version} REQUIRED)
+endif()
+
+if(NOT cudf_kafka_FOUND)
+  set(BUILD_TESTS OFF)
+  set(BUILD_BENCHMARKS OFF)
+  set(_exclude_from_all "")
+  if(CUDF_KAFKA_BUILD_WHEELS)
```

Comment:
I suspect a couple of the things in this block are superfluous, but since we're not actually building cudf_kafka wheels it's probably easiest to leave this block as is for now and make fixes later if/when we actually run and test it.